components/operator/internal/controllers/stack/stack_finalizer.go

```diff
@@ -0,0 +1,100 @@
+package stack
+
+import (
+	"context"
+	"time"
+
+	stackv1beta3 "github.com/formancehq/operator/apis/stack/v1beta3"
+	"github.com/formancehq/operator/internal/controllers/stack/delete"
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	finalizerName = "backup-delete"
+)
+
+var (
+	subjects = []string{"ledger", "payments"}
```

Comment:
Don't hardcode list of subject.
The registered module have a "NeedTopic" property.
This inform the core to create the topic.
You can retrieve it and dynamically retrieve the list of subjects